src/VestRewardsDistributionJob.sol

```diff
@@ -0,0 +1,178 @@
+
+        (address rewDist) = abi.decode(args, (address));
+
+        // prevent keeper from calling random contracts having distribute()
+        if (!distributions.contains(rewDist)) revert RewardDistributionDoesNotExist(rewDist);
+        // ensure that the right delay has elapsed
+        if (canDistributeAfter(rewDist) >= block.timestamp) revert CannotDistributeYet(rewDist);
+        // we omit checking the unpaid amount because if it is 0 it will revert during distribute()
+        uint256 distAmount = VestedRewardsDistributionLike(rewDist).distribute();
+        emit Work(network, rewDist, distAmount);
+    }
+
+    function workable(bytes32 network) external view override returns (bool, bytes memory) {
+        if (!sequencer.isMaster(network)) return (false, bytes("Network is not master"));
+
+        uint256 distributionsLen = distributions.length();
+        if (distributionsLen > 0) {
+            address distributable;
+            // this is used to find the distribute() than could have been called the earliest
+            // we use a hack that ensures the right functionality while avoiding an extra check for 0 value
+            uint256 earliestDistCall = block.timestamp + timeMagicNumber;
+            for (uint256 i = 0; i < distributionsLen; i++) {
+                address rewDist = distributions.at(i);
+                uint256 nextDistCall = canDistributeAfter(rewDist);
+                // timestamp should be strictly greater than nextDistCall
+                if (nextDistCall < block.timestamp) {
+                    uint256 vestId = VestedRewardsDistributionLike(rewDist).vestId();
+                    DssVestWithGemLike dssVest = VestedRewardsDistributionLike(rewDist).dssVest();
+                    uint256 amount = dssVest.unpaid(vestId);
+                    if (amount > 0){
+                        if (nextDistCall < earliestDistCall)
+                            distributable = rewDist;
+                    }
+                }
+            }
+            if (distributable == address(0)) return (false, bytes("No distribution"));
+            return (true, abi.encode(distributable));
+
+        }
+        else {
+            return (false, bytes("No farms"));
+        }
+    }
+
+    function rewardDistributionActive(address rewDist) public view returns (bool) {
+        return distributions.contains(rewDist);
+    }
+
+    function canDistributeAfter(address rewDist) internal view returns (uint256) {
+        // get the last time distribute() was called
+        uint256 distTimestamp = VestedRewardsDistributionLike(rewDist).lastDistributedAt();
+        if (distTimestamp == 0){
+            // first ditribution, we allow to be distributed immediately so there is deadlock
+            return block.timestamp - 1;
+        }
+        else{
+            // calculate when distribute() is allowed to be called next and return the value
+            return distTimestamp + distributionDelays[rewDist];
+        }
+    }
```

Comment:
Why do we need such logic?
Wouldn't it suffice to have the following check: `rewDist.lastDistributedAt() + distributionDelays[rewDist] < block.timestamp`?
It covers both the 1st and any subsequent calls.